src/__tests__/send-request.js

```diff
@@ -1,4 +1,72 @@
-import { encodePostData } from '../send-request'
+import { encodePostData, xhr } from '../send-request'
+
+const generateXmlHttpRequestMock = (status = 502) => ({
+    open: jest.fn(),
+    setRequestHeader: jest.fn(),
+    onreadystatechange: jest.fn(),
+    send: jest.fn(),
+    readyState: 4,
+    responseText: JSON.stringify('something here'),
+    status: status,
+})
+
+const generateXhrParams = (markRequestFailed, onXHRError = () => {}) => ({
+    url: 'https://any.posthog-instance.com',
+    data: '',
+    headers: {},
+    options: {},
+    captureMetrics: {
+        incr: () => {},
+        startRequest: () => {},
+        decr: () => {},
+        finishRequest: () => {},
+        markRequestFailed,
+    },
+    callback: () => {},
+    retriesPerformedSoFar: null,
+    retryQueue: {
+        enqueue: () => {},
+    },
+    onXHRError,
+})
+
+describe('when xhr requests fail', () => {
+    given('mockXHR', generateXmlHttpRequestMock)
+    given('markRequestFailed', jest.fn)
+
+    beforeEach(() => {
+        window.XMLHttpRequest = jest.fn(() => given.mockXHR)
+    })
+
+    it('does not error if the configured onXHRError is not a function', () => {
+        const onXHRError = {}
+        xhr(generateXhrParams(given.markRequestFailed, onXHRError))
+
+        expect(() => {
+            given.mockXHR.onreadystatechange()
+        }).not.toThrow()
```

Comment:
If I'm reading this correctly, this is checking that `(jest.fn())()` doesn't throw, which doesn't seem that useful...? But I'm probably reading it wrong.

Reply:
Ah, no, it checks it doesn't throw if the user configured onXHRError is not a function

So, someone could get confused and `onXHRError = 'I did not understand what to set here'` if we call without checking it is a function then we would throw. This way what used to be a fixed `console.error` can't start breaking the library (I think)

Reply:
Changed the test setup to see if it helps clarify

Reply:
Ah, I was just royally confused when following the trail that leads to actual code that runs when `mockXHR.onreadystatechange()` is called. Debugging helped, but now you are legitimately passing a function here, so the test is kind of... wrong?

![image](https://user-images.githubusercontent.com/53387/135709433-f05cba83-33d4-4f7e-8f0f-540ebc00b3cf.png)